### ISSUE TITLE
Soft hunger proposal

### ISSUE THREAD

**Issue by HybridDog** [minetest/minetest_game#2598]
I think that hunger should be added so that players do farming and not to make the gameplay more difficult or annoying; many hunger mods require the player to eat all the time and they kill the player if he/she does not eat, so with these mods I'm just dying and respawning because it's easier than collecting food (building a farm is not worth the effort).

Here's a proposal for a soft hunger mechanism which should make the player collect food (do farming) and harm the gameplay only after many hours of playing:
* Digestion of food takes 8 h realtime, i.e. it takes 8h from full to empty stomach.
  => The player can build a lot and does not constantly have to grab food, i.e. there is no gameplay-disturbing eating obligation where you always must have food in your inventory.
     Furthermore, eating affects the gameplay over a long time (e.g. play every day on server) instead of the gameplay during one hour cave exploration.
* If the player is hungry, the HP after respawn are low (e.g. 3 instead of 20).
  The number of HP is not necessarily linear wrt the hungriness.
  => Eating is optional, but beneficial if e.g. the player goes exploring dangerous areas.
    Low HP after respawning should be enough reason for the player to build a farm.
* Eating refills the HP and the stomach.
  => If the player is hurt, he/she can eat to refill his/her HP and at the same time refill his/her stomach.
* There is no automatic HP increasing or decreasing caused by hunger.
  => Food only affects gameplay if the player does something dangerous, i.e. it is soft hunger.

I've added this as new issue because #853 is very big and is, as far as I can see, about a canonical hunger mechanic whereas this is about a specific hunger mechanism.

**Comment by Df458** [comment]
Honestly, I don't think this suggestion makes any sense as presented. Here are my thoughts:

> If the player is hungry, the HP after respawn are low (e.g. 3 instead of 20).

This is a somewhat novel take on hunger, but I see several major issues with it.

1. Since eating after respawning would easily make up the difference in health, and building a home near spawn (or setting spawn in a home via beds) is common practice, _there's no reason for the average player to engage with this mechanic at all_.
2. This feels like a problematic positive feedback loop to me, players who have plenty to eat will be unlikely to die in the first place (because they can heal) and players who don't eat will be more predisposed to dying anyway.
3. Even if the other issues didn't exit, how would you suggest we communicate this state to the player? This is an important concern because hunger has no impact on the player's current state, and the player won't have a reason to eat if they don't know they're hungry to begin with. I suspect that there's no good answer here short of literally saying "hey you're hungry, if you die you'll respawn with low health" (which is a pretty terrible option)--either the player fears a danger that doesn't exist, or becomes confused and annoyed on respawn.

Finally, I've just gotta point this out

> Digestion of food takes 8 h realtime, i.e. it takes 8h from full to empty stomach.
=> The player can build a lot and does not constantly have to grab food, i.e. there is no gameplay-disturbing eating obligation where you always must have food in your inventory.

Nothing in the remainder of your suggestion obliges the player to eat unless they're deliberately putting themselves in danger. How often do you actually die when building? This seems like a very strange reason to make the player only eat roughly once every 3-4 ingame *weeks*, and looks more like a  knee-jerk reaction to other hunger systems than a well thought-out design decision.

**Comment by paramat** [comment]
See related #853 #855

**Comment by HybridDog** [comment]
> Since eating after respawning would easily make up the difference in health, […]

The player can only eat after respawning if he/she have food.

> […] and building a home near spawn (or setting spawn in a home via beds) is common practice, there's no reason for the average player to engage with this mechanic at all.

If the player walks away from the spawn and home to collect materials, he/she probably does not want to die and lose the items to a bones node.

> In other words, someone with no access to food is doubly-screwed and someone with food probably won't be affected at all.

This is intended. If the player does not collect food, he/she should be affected. Since hunger is so slow, the player has multiple hours to collect or create food.

> Even if the other issues didn't exist, how would you suggest we communicate this state to the player?

In the formspec where the player can click on respawn, we could show the current hungriness and how many HP after respawn are added due to this hungriness.

> How often do you actually die when building? 

I die very often because my buildings are big, so I often fall down when I haven't yet built the stairs. I don't know how many other people build like this too.

> Nothing in the remainder of your suggestion obliges the player to eat unless they're deliberately putting themselves in danger.

If there is a mobs mod on the server (or another mod which adds hurting things), there is always now and then a situation where the player looses some health, so there is always a bit of danger. 
This hunger mechanism should not oblige players to eat, but rather give them a benefit over not eating something. If eating some items keeps the hungriness low for many hours, eating something does not require many inventory item movements in the long run, thus it can be advantageous over eating nothing.

**Comment by Df458** [comment]
> If the player walks away from the spawn and home to collect materials, he/she probably does not want to die and lose the items to a bones node.

This is actually an argument *against* your proposal. The player already has a reason to bring food with them when they go out (for the reason your just stated), so hunger adds nothing of value in this context.

> In the formspec where the player can click on respawn, we could show the current hungriness and how many HP after respawn are added due to this hungriness.

That's too late though, at that point. If the player doesn't know about this mechanic *before* dying, then there's a good chance they won't have a handy source of food and setting that up will be much harder. This goes back to my comment about feedback loops: You're suggesting a punishment that's trivial for players who know about it, and unfair for players who don't.

> This hunger mechanism should not oblige players to eat, but rather give them a benefit over not eating something. If eating some items keeps the hungriness low for many hours, eating something does not require many inventory item movements in the long run, thus it can be advantageous over eating nothing.

But that's not true. There's no benefit here, only an obscure punishment. Adding a hunger mechanic that's not clear, engaging, and relevant is worse than adding no mechanic at all.

--

What seems to have spurred this suggestion is a pretty specific personal take on low health/hunger, where you

> ...kill myself to get full hp and hunger bar

To me, that doesn't imply the need for "soft" hunger--it implies that death is very cheap right now. Respawning with less health won't change that, because health is quick & easy to regain and the cost of healing from 3 health is cheaper than the cost of healing from 1 or 2. I suspect that you and other players will still kill yourselves because death is still cheap, the entire cost can be mitigated with a box of apples sitting near spawn.



Personally, I don't think a good hunger mechanic can be added to MTG without further systems that support it. Games are not banking software, you can't just slot in "features" and expect them to work from a design standpoint. Rather, they need a more complete creative vision in order to work.

To reinforce that, let's perform a quick thought experiment:

*Imagine we took Minecraft and made it so that hunger didn't kill the player. Despite this, the player would still have reason to eat: Health can't be (cheaply) restored without a full hunger bar. Note that the "killing yourself" method of healing would still come at a real cost as well, because most of a player's XP is lost on death and takes some time to regain.*

I think this demonstrates pretty well how adjacent systems must combine to make a single mechanic meaningful. Without the changes to healing, hunger would be a meaningless mechanic that everyone ignored. Without XP, death would be cheaper than (or at best, as expensive as) eating. It's also worth noting that this was not just an accident, the healing changes and addition of XP occurred simultaneously with the introduction of hunger.

**Comment by Df458** [comment]
You know... Honestly, I don't care any more. I've gotten tired of seeing the strange suggestions people come up with, and I don't have the patience to fight about designing a game like MTG correctly.

Y'all do what you want, I'm out.

**Comment by paramat** [comment]
> Digestion of food takes 8 h realtime

MT days are variable in length, default 20 mins, how food lasts should be scaled to the length of MT days, for example half of day length.
> If the player is hungry, the HP after respawn are low

I understand your reason, but this is weird. Death and respawn is a reset and should result in full health, otherwise it is not actually death.
> There is no automatic HP increasing or decreasing caused by hunger.

> Eating is optional,

No HP decrease from hunger? This is how harm should be caused, not by reduced HP after spawn. It seems wrong that if you do not die there is no harm from hunger. Much of survival gameplay does not involve death.
> many hunger mods require the player to eat all the time and they kill the player if he/she does not eat

Makes sense.
 > does not constantly have to grab food, i.e. there is no gameplay-disturbing eating obligation where you always must have food in your inventory.

That is the point of having hunger, it disturbs gameplay.
:-1:

**Comment by CandyFiend** [comment]
Hello everyone. I partially agree with HybridDog. I would like the players to use food, not a first-aid kit in the form of food, and do it more often. Therefore, I propose adding effects related to food. For starters, you can add a fine to the speed of movement and the extraction of blocks, not big, but so that in severe hunger it appears. This is a signal to the player that he is hungry and he needs to find food. If the player is well-fed, then he should get some kind of bonus, for example, to the speed of movement (I would add a sprint, the speed of which depends on hunger). The consumption of satiety should depend on the player’s actions, for example, if the player runs or gets blocks, his hunger is consumed faster (hunger will partially work as a stamina). Well, if a player ate, then for a short time he does not spend hunger (by itself, from running). In minecraft, hunger is spent immediately and steve is constantly forced to eat in order to maintain satiety at the maximum level. I would like to fix it. Well, so that players do not die, you need to make the hunger value with a respawn of 20-30% (at 0%, the player should take damage). This is enough to run home or find food, such as berries or something else. You can also add crafting mechanics or the like, which will spend hunger on the manufacture of expensive and complex objects or blocks. Using onions can also waste hunger when pulling a aunt. If it was possible to use 2 arm and shields were added, they could also use hunger. This should be discussed since Now I see no obvious reason to use food. The player must be shown that if he eats, it will give him advantages, but if he doesn’t, he will receive a fine, in the game there are such rules.
Sorry for the mistakes, I used an online translator.

**Comment by paramat** [comment]
Closing due to #2710
